dataops/components/resource_lock/views.py

```
@@ -0,0 +1,108 @@
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from common import LoggerFactory
+from fastapi import APIRouter
+from fastapi import Depends
+from fastapi_utils.cbv import cbv
+
+from dataops.components.resource_lock.crud import ResourceLockerCRUD
+from dataops.components.resource_lock.dependencies import get_resource_lock
+from dataops.components.resource_lock.schemas import ResourceLockBulkCreateSchema
+from dataops.components.resource_lock.schemas import ResourceLockBulkDeleteSchema
+from dataops.components.resource_lock.schemas import ResourceLockBulkResponseSchema
+from dataops.components.resource_lock.schemas import ResourceLockCreateSchema
+from dataops.components.resource_lock.schemas import ResourceLockResponseSchema
+from dataops.config import ConfigClass
+from dataops.dependencies import Cache
+from dataops.dependencies.cache import get_cache
+from dataops.models.base_models import EAPIResponseCode
+from dataops.resources.error_handler import catch_internal
+
+logger = LoggerFactory(
+    'resource_lock',
+    level_default=ConfigClass.LOG_LEVEL_DEFAULT,
+    level_file=ConfigClass.LOG_LEVEL_FILE,
+    level_stdout=ConfigClass.LOG_LEVEL_STDOUT,
+    level_stderr=ConfigClass.LOG_LEVEL_STDERR,
+).get_logger()
+
+router = APIRouter()
+
+
+@cbv(router)
+class RLock:
+    @router.post('/', response_model=ResourceLockResponseSchema, summary='Create a new lock')
+    @catch_internal('api_resource_lock')
+    async def lock(
+        self, body: ResourceLockCreateSchema, resource_locker: ResourceLockerCRUD = Depends(get_resource_lock)
+    ) -> ResourceLockResponseSchema:
+        unlocked = await resource_locker.perform_rw_lock(body.resource_key, body.operation)
+        response = ResourceLockResponseSchema(
+            code=EAPIResponseCode.success if unlocked else EAPIResponseCode.conflict, key=body.resource_key
```

we should raise an HTTP error with the correct code. 
like this one:
https://github.com/PilotDataPlatform/dataops/blob/develop/dataops/components/exceptions.py#L74

---

Fixed. Now raising exceptions instead from https://github.com/PilotDataPlatform/dataops/blob/develop/dataops/components/exceptions.py.